lms/djangoapps/courseware/grades.py

```diff
@@ -392,7 +392,9 @@ def _grade(student, course, keep_raw_scores):
 
     More information on the format is in the docstring for CourseGrader.
     """
-    course_structure = get_course_blocks(student, course.location)
+    course_structure = getattr(course, 'structure', None)
```

Comment:
Why choose to covertly pass the cached structure on the course object rather than passing it as a separate parameter?

I also wonder if there's a general place to cache the value so we can make use of it throughout LMS?  For example, if `get_course_blocks` itself caches the result in the request_cache?  This may be the longer-term approach for it.  In the meantime, passing it as a separate param within grading may be sufficient.